src/app/test/test.component.ts

```diff
@@ -30,18 +30,22 @@ export class TestComponent implements OnInit {
   };
   @Output() openCompareReportsEvent = new EventEmitter<any>();
   @ViewChild(CloneModalComponent) cloneModal!: CloneModalComponent;
-  @ViewChild(TestSettingsModalComponent)
-  testSettingsModal!: TestSettingsModalComponent;
-  @ViewChild(TestFolderTreeComponent)
-  testFolderTreeComponent!: TestFolderTreeComponent;
+  @ViewChild(TestSettingsModalComponent) testSettingsModal!: TestSettingsModalComponent;
   @ViewChild(DeleteModalComponent) deleteModal!: DeleteModalComponent;
+  @ViewChild(TestFolderTreeComponent) testFileTreeComponent!: TestFolderTreeComponent;
 
   constructor(
     private httpService: HttpService,
     private helperService: HelperService,
     private toastService: ToastService,
     private tabService: TabService,
-  ) {}
+  ) {
+    this.getGeneratorStatus();
```

Comment:
Could this be in OnInit?

Reply:
It can be, but it retrieves from localstorage so it's not necessary to call it in onInit

Reply:
I could say the same for calling it in the constructor instead of using the correct lifecycle method

Reply:
Yes however the method sets a class variable that is needed in the HTML, therefore getting it before the ngOnInit would be better as otherwise an `NG0100: Expression has changed after it was checked` error is thrown.

Reply:
Interesting, seems like angular is parsing the template before the component is initialized according to their own lifecycle? That is kind of disappointing, but I can see why you would call this in the constructor now